### PR TITLE
Fix periodic commit in server

### PR DESCRIPTION
**Pull request by davidegrohmann**
Periodic commit wasn't working in server due to the fact of nesting a
placebo transaction for executing the Cypher statement. This placebo
transaction was preventing periodic commit to actually commit in the
middle of the query.  This commit fixes that behaviour by executing
periodic commit in a toplevel transaction.

### CODE REVIEW

File: community/server/src/main/java/org/neo4j/server/rest/transactional/TransactionTerminator.java [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-    public TransactionTerminator( Transaction transaction )
-    {
-        this.transaction = transaction;
-    }
-
-    public void terminate()
-    {
-        transaction.terminate();
-    }
-}
```
Comment (henriknyman) [inline community/server/src/main/java/org/neo4j/server/rest/transactional/TransactionTerminator.java:37]: Terminate the terminator!

File: community/server/src/main/java/org/neo4j/server/rest/web/CypherService.java [hunk 2]
```diff
@@ -118,6 +130,11 @@ public Response cypher(String body,
                 includePlan = result.getQueryExecutionType().requestedExecutionPlanDescription();
             }
 
+            if ( periodicCommitQuery )
+            {
+                handler.setTransaction( database.beginTx() );
```
Comment (henriknyman) [inline community/server/src/main/java/org/neo4j/server/rest/web/CypherService.java:135]: Shouldn't we begin this transaction with the same type and AccessMode as the transaction we are closing above?

Comment (davidegrohmann) [inline community/server/src/main/java/org/neo4j/server/rest/web/CypherService.java:135]: Yes we should

Comment (davidegrohmann) [inline community/server/src/main/java/org/neo4j/server/rest/web/CypherService.java:135]: On the other hand, this transaction is used only for streaming back results and all the transactions opened for the `CypherService` have `AccessMode.Static.FULL`. So I don't think it is super important to fix it now, but we definitely improve it when we have security in place.

**Pull request merged**